diff --git a/package/dtc/dtc.hash b/package/dtc/dtc.hash
index 5856156f00..1c02573018 100644
--- a/package/dtc/dtc.hash
+++ b/package/dtc/dtc.hash
@@ -1,5 +1,6 @@
 # from https://www.kernel.org/pub/software/utils/dtc/sha256sums.asc
 sha256  29edce3d302a15563d8663198bbc398c5a0554765c83830d0d4c0409d21a16c4  dtc-1.7.0.tar.xz
+sha256  92d8ca769805ae1f176204230438fe52808f4e1c7944053c9eec0e649b237539  dtc-1.7.2.tar.xz
 
 # Hash for license files
 sha256  78e1e1ce5728a990d59c7de61d4741c49185d5129daf783b386380a214328d68  README.license
diff --git a/package/dtc/dtc.mk b/package/dtc/dtc.mk
index 5868ae0e5a..7dd21665bc 100644
--- a/package/dtc/dtc.mk
+++ b/package/dtc/dtc.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-DTC_VERSION = 1.7.0
+DTC_VERSION = 1.7.2
 DTC_SOURCE = dtc-$(DTC_VERSION).tar.xz
 DTC_SITE = https://www.kernel.org/pub/software/utils/dtc
 DTC_LICENSE = GPL-2.0+ or BSD-2-Clause (library)
@@ -22,6 +22,7 @@ DTC_MAKE_OPTS = \
 # For the host, we install headers in a special subdirectory to avoid
 # conflicts with the in-kernel libfdt copy.
 HOST_DTC_MAKE_OPTS = \
+	EXTRA_CFLAGS="-Wno-error=discarded-qualifiers -fPIC" \
 	PREFIX=$(HOST_DIR) \
 	INCLUDEDIR=$(HOST_DIR)/include/libfdt \
 	NO_PYTHON=1 \
